library/std/src/io/impls.rs

```diff
@@ -355,20 +355,31 @@ impl Write for &mut [u8] {
     }
 }
 
+const OOM_ERROR_MSG: &str =
+    "Out Of Memory. This panic may become an io::Error in the future (#84612)";
```

Comment:
I don't think we should include these details in here. This message will not be produced by the Rust compiler, but by any program written in Rust. Things like `io::Error` or `#84612` don't have a lot of meaning in that context. (E.g. imagine a web browser aborting with a message containing these terms. Users will not understand `#84612` to be a reference to the *Rust* issue tracker.)
```suggestion
    "out of memory";
```